### PR TITLE
releng/docs: refactor handbooks for RelEng

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this:
/kind cleanup
/kind documentation

#### What this PR does / why we need it:

Refactor RelEng documentation

#### Which issue(s) this PR fixes:

Related to https://github.com/kubernetes/sig-release/issues/1324

### CODE REVIEW

**Comment by cpanato** [comment]
Adding as a draft to collect some feedback on how I initially made the refactor. If this structure looks good I will move forward to the next phase which is to update the text

@puerco @saschagrunert @jeremyrickard @justaugustus 
@kubernetes/release-engineering

File: release-engineering/README.md [hunk 1]
```diff
@@ -208,6 +209,13 @@ that any final cherry-pick content is merged ahead of that last release.
...
-  - Maintaining instructions on how to consume all above official artifacts, plus variations for how 
+  - Maintaining instructions on how to consume all above official artifacts, plus variations for how
 consumers can test pre-release Alpha/Beta/RC artifacts
   - push-build.sh maintenance, until it goes away
+
+### Handbooks
+
+- [Release Managers Onboarding](./handbooks/onboarding.md)
+- [Release Management Hanbook](./handbooks/release-management.md)
+- [Branch Management Hanbook](./handbooks/branch-manager.md)
```
Comment (saschagrunert) [inline release-engineering/README.md:220]:
```suggestion
- [Release Management Handbook](./handbooks/release-management.md)
- [Branch Management Handbook](./handbooks/branch-manager.md)
```

**Review by saschagrunert — COMMENTED** [review]
I think the intention is right. We still would have to find better names for "Release Management" and "Branch management" from my point of view. How about:

- Onboarding and Tooling
- Release Cycle Handbook
- Patch Management Handbook

File: release-engineering/README.md (hunk 1 above)
Comment (cpanato) [reply release-engineering/README.md:220]: Thanks for the feedback Sascha, I will continue the work and make the changes you suggested

**Comment by cpanato** [comment]
@saschagrunert @puerco did some changes and refactor a bit more, please take a look when you have some time and lets discuss any changes

**Comment by justaugustus** [comment]
@cpanato -- can you give this a rebase?

**Comment by cpanato** [comment]
> @cpanato -- can you give this a rebase?

missed this notification, will do

**Comment by cpanato** [comment]
@saschagrunert @puerco let me know if you think this is a good starting point, then I will rebase and maybe we can merge this to follow up where needed

**Comment by saschagrunert** [comment]
@cpanato yes let's rebase and move forward with this one :+1:

**Pull request closed without merging**